client/ayon_core/lib/transcoding.py

```diff
@@ -834,6 +834,112 @@ def get_ffprobe_streams(path_to_file, logger=None):
+                "framespersecond" and evaluated to float value.
+        logger (logging.Logger): Logger used for logging.
+    """
+    if logger is None:
+        logger = logging.getLogger(__name__)
+
+    def _ffprobe_metadata_conversion(metadata):
+        """Convert ffprobe metadata unified format."""
+        output = {}
+        for key, val in metadata.items():
+            if key in ("tags", "disposition"):
+                output.update(val)
+            else:
+                output[key] = val
+        return output
+
+    metadata_stream = None
+    ext = os.path.splitext(path_to_file)[-1].lower()
+    if ext not in IMAGE_EXTENSIONS:
+        logger.info((
+            "File extension \"{}\" is not supported by OpenImageIO."
+            " Trying to get metadata using FFprobe."
+        ).format(ext))
+        ffprobe_stream = get_ffprobe_data(path_to_file, logger)
+        if "streams" in ffprobe_stream and len(ffprobe_stream["streams"]) > 0:
+            metadata_stream = _ffprobe_metadata_conversion(
+                ffprobe_stream["streams"][0])
+
+    if not metadata_stream and is_oiio_supported():
+        oiio_stream = get_oiio_info_for_input(path_to_file, logger=logger)
+        if "attribs" in (oiio_stream or {}):
+            metadata_stream = {}
+            for key, val in oiio_stream["attribs"].items():
+                if "smpte:" in key.lower():
+                    key = key.replace("smpte:", "")
+                metadata_stream[key.lower()] = val
+            for key, val in oiio_stream.items():
+                if key == "attribs":
+                    continue
+                metadata_stream[key] = val
+    else:
+        logger.info((
+            "OpenImageIO is not supported on this system."
+            " Trying to get metadata using FFprobe."
+        ))
+        ffprobe_stream = get_ffprobe_data(path_to_file, logger)
+        if "streams" in ffprobe_stream and len(ffprobe_stream["streams"]) > 0:
+            metadata_stream = _ffprobe_metadata_conversion(
+                ffprobe_stream["streams"][0])
+
+    if not metadata_stream:
+        logger.warning("Failed to get metadata from image file.")
+        return {}
+
+    if keys is None:
+        return metadata_stream
+
+    # create framerate key from available ffmpeg:r_frame_rate
+    # or oiiotool:framespersecond and evaluate its string expression
+    # value into flaot value
+    if (
+        "r_frame_rate" in metadata_stream
+        or "framespersecond" in metadata_stream
+    ):
+        rate_info = metadata_stream.get("r_frame_rate")
+        if rate_info is None:
+            rate_info = metadata_stream.get("framespersecond")
+
+        try:
+            metadata_stream["framerate"] = eval(str(rate_info))
+        except Exception as e:
+            logger.warning(
+                "Failed to evaluate '{}' value to framerate. Error: {}".format(
+                    rate_info, e)
+            )
+
+    output = {}
+    for key in keys:
```

Comment:
What should this loop do?

Reply:
Aggregate all required metadata from prepared metadata stream

Reply:
Why it does combine it with nested keys?

Reply:
in some cases there could be nested dictionaries where the key should be search in

Reply:
hey @iLLiCiTiT could we just merge this?

Reply:
Linking comment that is the most important https://github.com/ynput/ayon-core/pull/44#discussion_r1695458171

Reply:
have moved it into the loader plugin and improved docstring. [`ede1b17` (#44)](https://github.com/ynput/ayon-core/pull/44/commits/ede1b1737013df3b8f0a38d5d781511e9915d10c)